src/main/java/org/elasticsearch/index/query/TermQueryParser.java

```diff
@@ -66,9 +60,9 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                     currentFieldName = parser.currentName();
                 } else {
                     if ("term".equals(currentFieldName)) {
-                        value = parser.objectBytes();
+                        value = parser.objectText();
```

Comment:
Also would like feedback on this change: Did this for `fromXContent()` tests to work. When value field is String, was read in here as BytesRef (internally call to JsonXContentparser#utf8Bytes()). This would lead to non-equal queries for tests, original value being String, after `fromXContent` BytesRef. Again, this shouldn't matter once lucene query is constructed. I changed to `objectText()` here, alternatively I could convert back for equality test to work, but that seems awkward.
```
value = parser.objectBytes();
if (value instanceof BytesRef) {
   value = ((BytesRef) value).utf8ToString();
}
```
Thoughts?